Source/Core/DolphinQt2/Config/GeckoCodeWidget.cpp

```diff
@@ -0,0 +1,205 @@
+// Copyright 2017 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include "DolphinQt2/Config/GeckoCodeWidget.h"
+
+#include <QFontDatabase>
+#include <QFormLayout>
+#include <QLabel>
+#include <QListWidget>
+#include <QMessageBox>
+#include <QPushButton>
+#include <QTextEdit>
+#include <QVBoxLayout>
+
+#include "Common/FileUtil.h"
+#include "Common/IniFile.h"
+#include "Core/ConfigManager.h"
+#include "Core/GeckoCodeConfig.h"
+#include "DolphinQt2/Config/CheatWarningWidget.h"
+#include "DolphinQt2/GameList/GameFile.h"
+
+GeckoCodeWidget::GeckoCodeWidget(const GameFile& game)
+    : m_game(game), m_game_id(game.GetGameID().toStdString()), m_game_revision(game.GetRevision())
+{
+  CreateWidgets();
+  ConnectWidgets();
+
+  auto game_ini_default = SConfig::GetInstance().LoadDefaultGameIni(m_game_id, m_game_revision);
+  auto game_ini_local = SConfig::GetInstance().LoadLocalGameIni(m_game_id, m_game_revision);
+
+  m_gecko_codes = Gecko::LoadCodes(game_ini_default, game_ini_local);
+
+  UpdateList();
+}
+
+void GeckoCodeWidget::CreateWidgets()
+{
+  auto* layout = new QVBoxLayout;
+  m_warning = new CheatWarningWidget(m_game_id);
+  m_code_list = new QListWidget;
+  m_name_label = new QLabel;
+  m_creator_label = new QLabel;
+  m_code_description = new QTextEdit;
+  m_code_view = new QListWidget;
+  m_download_codes = new QPushButton(tr("Download Codes (WiiRD Database)"));
+
+  m_code_description->setReadOnly(true);
+
+  m_download_codes->setEnabled(!m_game_id.empty());
+
+  layout->addWidget(m_warning);
+  layout->addWidget(m_code_list, 3);
+
+  auto* info_layout = new QFormLayout;
```

Comment:
On macOS, this centers the labels, which 1) looks weird, since it's not a form dialog, and 2) jumps around as the value text changes width. Since macOS allows what Qt calls the Plastique (or modern KDE) style in information presentation situations (e.g., Finder's "Get Info" window), I think the appropriate thing to do here is `setFormAlignment(Qt::AlignLeft | Qt::AlignTop)`.